### PR TITLE
[KYUUBI #6489] [PYTHON] PyKyuubi get_table_names also supports Spark SQL dialect

### LINKED ISSUE #6489: [Bug] function 'get_table_names' in the python client returns incorrect results
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

image
image

Hi,
I encountered the problem in the above image when I tried to connect Kyuubi with Superset.
The table structure and table name are incorrect, and appear to return a schema.

It's not what I want.

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

```yaml
default
```

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by BruceWong96**
# :mag: Description
## Issue References 🔗

This pull request fixes #6489

## Describe Your Solution 🔧

After my investigation, I found the bug and solution.
The function get_table_names returns an incorrect value when I used Superset to connect to Kyuubi for Spark SQL.
[get_table_names](https://github.com/apache/kyuubi/blob/master/python/pyhive/sqlalchemy_hive.py#L380)

The following code is used to connect to hive directly.
`return [row[0] for row in connection.execute(text(query))]` 

Because The following value is returned when the Hive is connected.

show tables in default :
[('student',), ('student_scores',)]

The following code is used to connect to Kyuubi.
`return [row[1] for row in connection.execute(text(query))]` 

Because The following value is returned when the Kyuubi is connected.

show tables in default :
[('default', 'employees', False), ('default', 'student', False), ('default', 'student_scores', False)]

So, for the difference in return value, I modified the code.

And I test them in Superset. The code works.

Hive
image

Kyuubi
image

Spark SQL also works properly.
image

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: python/pyhive/sqlalchemy_hive.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def get_table_names(self, connection, schema=None, **kw):
         query = 'SHOW TABLES'
         if schema:
             query += ' IN ' + self.identifier_preparer.quote_identifier(schema)
-        return [row[0] for row in connection.execute(text(query))]
+        return [row[1] for row in connection.execute(text(query))]
```
Comment (pan3793) [inline python/pyhive/sqlalchemy_hive.py:380]: this breaks the Hive case.

The PyKyuubi is supposed to work with Kyuubi, Spark Thrift Server, and HiveServer2. In this case, we may need to either switch to the Metadata Thrift API or create different dialects

Comment (BruceWong96) [reply python/pyhive/sqlalchemy_hive.py]: > this breaks the Hive case.
> 
> The PyKyuubi is supposed to work with Kyuubi, Spark Thrift Server, and HiveServer2. In this case, we may need to either switch to the Metadata Thrift API or create different dialects

Thank you for your reply. 
I have made modifications and tests. 
The specific details are in the issue.

File: python/pyhive/sqlalchemy_hive.py [hunk 2]
```diff
@@ -377,7 +377,17 @@ def get_table_names(self, connection, schema=None, **kw):
...
             query += ' IN ' + self.identifier_preparer.quote_identifier(schema)
-        return [row[1] for row in connection.execute(text(query))]
+
+        table_names = []
+
+        for row in connection.execute(text(query)):
+            # Hive returns 1 columns
+            if len(row) == 1:
+                table_names.append(row[0])
+            # Kyuubi returns 3 columns
```
Comment (pan3793) [inline python/pyhive/sqlalchemy_hive.py:387]: actually, for SQL cases, Kyuubi's behavior totally depends on the computing engine
```suggestion
            # Spark SQL returns 3 columns
```

File: python/pyhive/sqlalchemy_hive.py [hunk 3]
```diff
@@ -377,7 +377,17 @@ def get_table_names(self, connection, schema=None, **kw):
...
+            elif len(row) == 3:
+                table_names.append(row[1])
```
Comment (pan3793) [inline python/pyhive/sqlalchemy_hive.py:389]: can we have an else branch to handle the other cases?

Comment (BruceWong96) [reply python/pyhive/sqlalchemy_hive.py:389]: > can we have an else branch to handle the other cases?

So is it better for the else branch to throw an exception or return UNKNOWN as a tablename ?

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**